packages/compiler/src/experimental/type-factory.ts

```diff
@@ -0,0 +1,313 @@
+import { Numeric } from "../core/numeric.js";
+import { Program } from "../core/program.js";
+import {
+  BooleanLiteral,
+  DecoratorApplication,
+  DecoratorFunction,
+  Model,
+  ModelProperty,
+  Namespace,
+  NumericLiteral,
+  RekeyableMap,
+  Scalar,
+  StringLiteral,
+  Type,
+  Union,
+  UnionVariant,
+} from "../core/types.js";
+import { createRekeyableMap } from "../utils/misc.js";
+import { Realm } from "./realm.js";
+type DecoratorArgs = DecoratorFunction | [DecoratorFunction, ...any[]];
+
+interface ScalarOptions {
+  extends?: Scalar;
+  namespace?: Namespace;
+}
+
+interface ModelOptions {
+  extends?: Model;
+  namespace?: Namespace;
+}
+
+interface ModelPropertyOptions {
+  optional?: boolean;
+}
+
+interface UnionVariantOptions {
+  union?: Union;
+}
+
+export function createTypeFactory(program: Program, realm?: Realm) {
```

Comment:
could we define the explicit interface for that ?

Reply:
Removed the type-factory in favor of typekits